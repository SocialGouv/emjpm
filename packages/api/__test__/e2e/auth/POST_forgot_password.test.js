//

const request = require("supertest");

// Mock nodemailer before importing the server !
const nodemailerMock = require("nodemailer-mock");
jest.setMock("nodemailer", nodemailerMock);
jest.setMock("rand-token", { uid: jest.fn(() => "kikoulol") });

// Fake env
process.env.SMTP_FROM = "ne-pas-repondre@emjpm.gouv.fr";
process.env.APP_URL = "https://emjpm.gouv.fr";

const server = require("@emjpm/api/app");
const knex = require("@emjpm/api/db/knex");

beforeEach(() => {
  nodemailerMock.mock.reset();
});

afterAll(async () => {
  await knex.destroy();
});

test("send a forgot password email to ud@ud.com with correct token on request", async () => {
  const response = await request(server)
    .post("/auth/forgot_password")
    .send({
      email: "ud@ud.com"
    });
  expect(response.body).toMatchInlineSnapshot(`""`);
  expect(nodemailerMock.mock.sentMail()).toMatchSnapshot();
  expect(response.status).toBe(200);
});

test("fail because udxxxx@ud.com is an invalid user", async () => {
  const response = await request(server)
    .post("/auth/forgot_password")
    .send({
      email: "udxxxx@ud.com"
    });
  expect(response.body).toMatchInlineSnapshot(
    { stack: expect.any(String) },
    `
Object {
  "message": "User \\"udxxxx@ud.com\\" not Found",
  "name": "NotFoundError",
  "stack": Any<String>,
  "status": 404,
}
`
  );
  expect(nodemailerMock.mock.sentMail()).toEqual([]);
  expect(response.status).toBe(404);
});
