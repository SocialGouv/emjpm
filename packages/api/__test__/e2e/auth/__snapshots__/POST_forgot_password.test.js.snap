// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`send a forgot password email to ud@ud.com with correct token on request 2`] = `
Array [
  Object {
    "from": "ne-pas-repondre@emjpm.gouv.fr",
    "html": "
Bonjour,<br>
<br>
Vous avez demandé le changement de mot de passe.
<br><br>
Veuillez cliquer sur le lien suivant: <a href=https://emjpm.gouv.fr/reset-password?token=kikoulol>réinitialiser mon mot de passe</a>.
<br><br><br>
Bien à vous.
",
    "subject": "Nouveau mot de passe pour e-MJPM",
    "text": "
Bonjour,

Vous avez demandé le changement de mot de passe.


Veuillez cliquer sur le lien suivant: https://emjpm.gouv.fr/reset-password?token=kikoulol

Bien à vous.
",
    "to": "ud@ud.com",
  },
]
`;
